### DISCUSSION TITLE
Has anyone combined a briefcase iOS project with SwiftUI views?

### DISCUSSION THREAD

**Discussion by radcli14** [beeware/briefcase#674] [Q&A]
I’ve built a skeleton of my iOS app using SwiftUI, and want to have some python code run underneath. I’ve tried a couple of different options that I found on GitHub, with some success, but not quite everything I need. Briefcase looks pretty well streamlined, but it’s iOS support is in ObjC. I’ve read that there are ObjC methods that will let you show a SwiftUI view. Anyone have any success combining these?

**Comment by freakboy3742** [comment]
As far as I know, it *should* be possible; however, I can't point you at an easy "how to" guide.

The best advice I can give you is that Swift and ObjC are *broadly* compatible. If you can find a guide on how to do this in Swift, the mapping from the Swift "spelling" of an API and the ObjC "spelling" is generally straightforward; so it may be possible to invoke the Swift API directly by doing no more than adjusting the syntax. However, I haven't tried this, so it will be a bit of an experiment on your part.
